reflect.go

```diff
@@ -149,6 +149,10 @@ type Reflector struct {
 	// include a schema ID.
 	BaseSchemaID ID
 
+	// UseCamelCaseForIDs determines if missing IDs will use CamelCase (e.g. `TestUser`)
+	// or the default snake-case (e.g. `test-user`) when creating the BaseSchemaID.
+	UseCamelCaseForIDs bool
```

Comment:
Rather than a specific case, It'd probably be better to use a specific method, which is consistent with some of the other text modifiers.

Something like:

```
SchemaIDNameFormatter func(id string) string
```

Reply:
OK, is this more like what you are thinking of, @samlown ?